### PR TITLE
KT scan tests generation through CMakeLists.txt

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we implementing KT scan tests generation through `CMakeLists.txt` :
- right now supported only next combinations in `CMakeLists.txt` :
```
TEST_DATA_PER_WORK_ITEM : 1, 2, 4, 8, 16, 32
TEST_WORK_GROUP_SIZE    : 64, 128, 256, 512, 1024
TEST_TYPE               : uint32_t
```

### CODE REVIEW

File: test/kt/CMakeLists.txt [hunk 1]
```diff
@@ -124,7 +124,49 @@ if (ONEDPL_TEST_ENABLE_KT_ESIMD)
     _generate_esimd_sort_test("256" "32" "double" "" 1000) # segfault
 endif()
 
+function (_generate_scan_test _data_per_work_item _work_group_size _type)
```
Comment (dmitriy-sobolev) [inline test/kt/CMakeLists.txt:127]: Let's rename it to `_generate_igpu_scan_test` to highlight the KT backend.

File: test/kt/CMakeLists.txt [hunk 2]
```diff
@@ -124,7 +124,49 @@ if (ONEDPL_TEST_ENABLE_KT_ESIMD)
...
+
+    if ((NOT TARGET build-scan-tests) AND (NOT TARGET run-scan-tests))
+        add_custom_target(build-scan-tests COMMENT "Build all scan kernel template tests")
```
Comment (dmitriy-sobolev) [inline test/kt/CMakeLists.txt:130]: Let's rename the umbrella targets to match the KT backend:
 - build-scan-tests -> build-igpu-tests
 - run-scan-tests -> run-igpu-tests

File: test/kt/CMakeLists.txt [hunk 3]
```diff
@@ -124,7 +124,49 @@ if (ONEDPL_TEST_ENABLE_KT_ESIMD)
...
+        add_custom_target(run-scan-tests
+            COMMAND "${CMAKE_CTEST_COMMAND}" -R "^run-scan-tests$" --output-on-failure --no-label-summary
+            DEPENDS build-scan-tests
+            COMMENT "Build and run all scan kernel template tests")
+    endif()
+
+    string(REPLACE "_t" "" _type_short ${_type})
+    set(_target_name "single_pass_scan_dpwi${_data_per_work_item}_wgs${_work_group_size}_${_type_short}")
```
Comment (dmitriy-sobolev) [inline test/kt/CMakeLists.txt:138]: single_pass_scan -> igpu_scan

File: test/kt/CMakeLists.txt [hunk 4]
```diff
@@ -124,7 +124,49 @@ if (ONEDPL_TEST_ENABLE_KT_ESIMD)
...
+        target_compile_definitions(${_target_name} PRIVATE TEST_DATA_PER_WORK_ITEM=${_data_per_work_item})
+        target_compile_definitions(${_target_name} PRIVATE TEST_WORK_GROUP_SIZE=${_work_group_size})
+        target_compile_definitions(${_target_name} PRIVATE TEST_TYPE=${_type})
+    endif()
+
+endfunction()
+
+function(_generate_scan_tests)
+    set(_data_per_work_item_all "8")
+    set(_work_group_size_all "256")
```
Comment (dmitriy-sobolev) [inline test/kt/CMakeLists.txt:156]: Basing on discussion on the main PR, we can expand the list to:

```suggestion
    set(_data_per_work_item_all "1" "2" "4" "8" "16" "32")
    set(_work_group_size_all "64" "128" "256" "512" "1024")
```

File: test/kt/single_pass_scan.cpp [hunk 5]
```diff
@@ -13,6 +13,9 @@
 //
 //===----------------------------------------------------------------------===//
 
+#define LOG_TEST_INFO 1
```
Comment (dmitriy-sobolev) [inline test/kt/single_pass_scan.cpp:16]: I guess it was left unintentionally .

File: test/kt/CMakeLists.txt [hunk 6]
```diff
@@ -124,7 +124,49 @@ if (ONEDPL_TEST_ENABLE_KT_ESIMD)
...
+    set(_type_all "int" "uint32_t")
```
Comment (dmitriy-sobolev) [inline test/kt/CMakeLists.txt:157]: `int` is not supported, so let's leave `uint32_t` only.

Comment (SergeyKopienko) [reply test/kt/CMakeLists.txt:157]: Fixed

File: test/kt/CMakeLists.txt (hunk 4 above)
Comment (SergeyKopienko) [reply test/kt/CMakeLists.txt]: Fixed

File: test/kt/CMakeLists.txt (hunk 3 above)
Comment (SergeyKopienko) [reply test/kt/CMakeLists.txt:138]: It's named now according to cpp filename

File: test/kt/CMakeLists.txt (hunk 2 above)
Comment (SergeyKopienko) [reply test/kt/CMakeLists.txt]: Fixed

File: test/kt/CMakeLists.txt (hunk 1 above)
Comment (SergeyKopienko) [reply test/kt/CMakeLists.txt]: Fixed

**Review by dmitriy-sobolev — APPROVED** [review]
LGTM.

You can also wait the confirmation from @adamfidel.

**Pull request merged**